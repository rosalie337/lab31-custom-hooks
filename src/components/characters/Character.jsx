import React from 'react'
import PropTypes from 'prop-types'

function Characters({name, url}) {
    return (
        <>
            <figure>
                <img src={url} />
                <figcaption>{name}</figcaption>  
            </figure>
        </>
    )
}

Characters.propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,

}

export default Characters

