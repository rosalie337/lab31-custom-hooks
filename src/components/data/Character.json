[
    {
      "_id": "5da237699734fdcb7bef8f56",
      "name": "Miles Shortman",
      "image": "https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
    }
]