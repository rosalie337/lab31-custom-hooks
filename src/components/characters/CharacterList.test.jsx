/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Characters component', () => {
  afterEach(() => cleanup());
  it('renders Characters', () => {
    const { asFragment } = render(<Character
      characters={[{
        name: 'Arik',
        url: 'https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002'
      }]}
    />);
      
    expect(asFragment()).toMatchSnapshot();
  });
});