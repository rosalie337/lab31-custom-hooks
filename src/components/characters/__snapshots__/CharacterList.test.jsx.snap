// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders Characters 1`] = `
<DocumentFragment>
  <div>
    <figure>
      <img />
      <figcaption />
    </figure>
  </div>
</DocumentFragment>
`;
